features/fake_time.feature

```diff
@@ -0,0 +1,129 @@
+Feature: Fake Time
+  Background:
+    Given a file named "features/support/hooks.js" with:
+    """
+    var sinon = require('sinon');
+    var hooks = function () {
+
+      this.Before({tags: ['@fakedTime']}, function(scenario, callback) {
+        this.clock = sinon.useFakeTimers();
+        callback();
+      });
+
+      this.After({tags: ['@fakedTime']}, function(scenario, callback) {
+        this.clock.restore();
+        callback();
+      });
+
+    };
+
+    module.exports = hooks;
+    """
+
+  Scenario: Use fake timers to faked time
+    Given a file named "features/passing_steps.feature" with:
+      """
+      Feature: a feature
+        @fakedTime
+        Scenario: a scenario
+          Given a faked time step
+      """
+
+    Given a file named "features/step_definitions/passing_steps.js" with:
+      """
+      var assert = require('assert');
+      var sinon = require('sinon')
+
+      stepDefinitions = function() {
+        this.Given(/^a faked time step$/, function (done) {
+          var testFunction = sinon.stub();
+          setTimeout(testFunction, 100);
+          assert(!testFunction.called);
+          this.clock.tick(101);
+          assert(testFunction.called);
+          done()
+        });
+      };
+
+      module.exports = stepDefinitions
+      """
+
+      When I run cucumber.js with `--strict`
+      Then it passes
+
+
+  Scenario: faked time doesn't trigger the test runner timeout
+
+    Given a file named "features/passing_steps.feature" with:
+      """
+      Feature: a feature
+        @fakedTime
+        Scenario: a scenario
+          Given a faked time step
+      """
+
+    Given a file named "features/step_definitions/passing_steps.js" with:
+      """
+      var assert = require('assert');
+      var sinon = require('sinon')
+
+      stepDefinitions = function() {
+        this.Given(/^a faked time step$/, function (done) {
+          var testFunction = sinon.stub();
+          setTimeout(testFunction, 10000);
+          assert(!testFunction.called);
+          this.clock.tick(10001);
+          assert(testFunction.called);
+          done()
+        });
+      };
+
+      module.exports = stepDefinitions
+      """
+      When I run cucumber.js with `--strict`
+      Then it passes
+
+
+  Scenario: real time is restored
```

Comment:
Yeah, but to cucumber-js real time was never faked. To me this is just testing that you correctly ran this.clock.restore() in your after hook. Not running that doesn't effect whether or not cucumber-js works.